src/hivpy/hiv_status.py

```diff
@@ -416,3 +441,145 @@ def disease_and_diagnosis(disease_col, diagnosis_col, disease_rate, diagnosis_pr
         HIV_deaths = r_death < prob_death
         self.output.record_HIV_deaths(pop, HIV_deaths)
         return HIV_deaths
+
+    def update_HIV_diagnosis(self, pop: Population):
+        """
+        Diagnose people that have been tested this time step. The default test type used
+        is Ab, but certain policy options make use of PCR (RNA VL) or Ag/Ab tests.
+        Accuracy depends on test sensitivity, PrEP usage, as well as CD4 count.
+        """
+        # tested population in primary infection
+        primary_pop = pop.get_sub_pop([(col.IN_PRIMARY_INFECTION, op.eq, True),
+                                       (col.LAST_TEST_DATE, op.eq, pop.date)])
+
+        if len(primary_pop) > 0:
+            # primary infection diagnosis outcomes
+            diagnosed = pop.transform_group([pop.get_correct_column(col.PREP_TYPE, dt=0),
+                                             pop.get_correct_column(col.PREP_TYPE, dt=1)],
+                                            self.calc_primary_diag_outcomes, sub_pop=primary_pop)
```

Comment:
Requires discussion: this part seems more difficult to remove use of last time step from.

Reply:
Add a variable to the PrEP module to keep track of when PrEP usage began.

Reply:
This is just to check that a person is taking injectable prep _and_ they started taking it before this timestep, so we would just need to check that the injectable prep start date is less than the current date.

Reply:
If a person is diagnosed with HIV then they do not go onto injectable prep, but the injectable prep code comes before the diagnosis because it affects sensitivity. As a result, we don't change the sensitivity if the injectable prep variable has only been set _this_ timestep, so that it can be reversed later and be self consistent.